### PR TITLE
runtime, state, core, consensus, cmd: Migrate runtime calls

### LINKED ISSUE #458: Migrate Runtime Calls
We need to move the majority of logic from the storage funcs in `runtime/imports.go` into our storage API. The storage API should be called from the imports in place of the existing logic.  

To support this we should instantiate a Trie inside the state service. For the time being lets leave the existing Trie alone.

By the end of this the runtime should depend on the storage API and not the Trie. We may beed the leave the existing Trie in there for now so we don't break tests (TBD).

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes
- Refactor to update references to `trie.Trie` to `state.StorageState`

## Tests:
```
go test -v ./...
```

### Issues:
closes #458

### CODE REVIEW

File: state/storage.go [hunk 1]
```diff
@@ -2,53 +2,92 @@ package state
...
 
-func (s *storageState) StorageRoot() (common.Hash, error) {
+func (s *StorageState) StorageRoot() (common.Hash, error) {
 	return s.trie.Hash()
 }
 
-func (s *storageState) EnumeratedTrieRoot(values [][]byte) {
+func (s *StorageState) EnumeratedTrieRoot(values [][]byte) {
 	//TODO
 }
```
Comment (ansermino) [inline state/storage.go]: please add a `panic("not implemented")` for this and `ClearPrefix`

File: runtime/wasmer.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ func (r *Runtime) Stop() {
 }
 
 func (r *Runtime) StorageRoot() (common.Hash, error) {
-	return r.trie.Hash()
+	return r.storage.StorageRoot()
 }
```
Comment (ansermino) [inline runtime/wasmer.go]: This should be removed once `core` is refactored, please can you add a `TODO`

**Comment by edwardmack** [comment]
Created interface `RuntimeStorage` in `runtime/interface.go` and updated code to use new interface, so runtime no longer relies on `state` package.

File: runtime/interface.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+package runtime
+
+import (
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/trie"
+)
+
+type RuntimeStorage interface {
```
Comment (ansermino) [inline runtime/interface.go]: Please rename to `Storage`, cardinal sin to include package name :wink:

**Pull request merged**